File: hw/astudillo/U3/HW18Factory/AbstractFactory/src/main/java/ec/edu/espe/composite/abstractfactory/controller/GUIFactory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.composite.abstractfactory.controller;

import ec.edu.espe.composite.abstractfactory.model.Button;
import ec.edu.espe.composite.abstractfactory.model.Menu;

/**
 *
 * @author PC
 */
public abstract class GUIFactory {
    
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
        if (sys == 0) {
            return new WinFactory();
        } else {
            return new LinuxFactory();
        }
    }
    
    private static int readFromConfigFile(String key) {
        // Simplified config reading - returns 0 for Windows, 1 for Linux
        String os = System.getProperty("os.name").toLowerCase();
        if (os.contains("win")) {
            return 0;
        } else {
            return 1;
        }
    }
    
    public abstract Button createButton();
    public abstract Menu createMenu();
}

File: hw/fuentes/U3/HW19ABSTRACT/HWABSTRACT/ec.edu.espe.client_AbstractFactory_jar_1.0-SNAPSHOT/HW19ABSTRACTPY/ec/edu/espe/client/model/GUIFactory.py
import platform

class GUIFactory:
    @staticmethod
    def getFactory():
        sys = GUIFactory.__readFromConfigFile("OS_TYPE")
        if sys == 0:
            from model.WinFactory import WinFactory
            return WinFactory()
        else:
            from model.LinuxFactory import LinuxFactory
            return LinuxFactory()

    @staticmethod
    def __readFromConfigFile(key):
        os_name = platform.system().lower()
        if "win" in os_name:
            return 0
        else:
            return 1

    def createButton(self):
        raise NotImplementedError

    def createMenu(self):
        raise NotImplementedError
